test/diagonal.jl

```diff
@@ -0,0 +1,205 @@
+using Snowflake
+using Test
+using StaticArrays
+
+@testset "diagonal_gate" begin
```

Comment:
This test group is getting a bit large. I think we should split it into pi8_gate and phase_gate